### PR TITLE
Adding new api endpoint (/vps-stats)

### PR DESCRIPTION
**Pull request by InvalidLenni**
- VPS classes are defined as ``classes.VPS``
- API Route defined as ``get_vps_stats()``

### CODE REVIEW

File: luxurynitro/api.py [hunk 1]
```diff
@@ -79,6 +79,20 @@ async def get_user(self) -> classes.User:
             ]
         )
     
+    async def get_vps_stats(self) -> list[classes.VPS]:
+        "Fetches the vps stats. Returns `vps`."
```
Comment (itschasa) [inline luxurynitro/api.py:83]:
```suggestion
        "Fetches the vps stats. Returns list of `VPS`."

```

File: luxurynitro/classes.py [hunk 2]
```diff
@@ -78,6 +85,7 @@ class Stats:
     support_time: int
     total_claims: int
 
+
```
Comment (itschasa) [inline luxurynitro/classes.py:88]:
```suggestion

```

File: luxurynitro/api.py [hunk 3]
```diff
@@ -79,6 +79,20 @@ async def get_user(self) -> classes.User:
             ]
         )
     
+    async def get_vps_stats(self) -> list[classes.VPS]:
+        "Fetches the vps stats. Returns `vps`."
+        res = await self.client.get('/vps-stats')
+        resjson = res.json()
+
```
Comment (itschasa) [inline luxurynitro/api.py:86]:
```suggestion

data = sorted(resjson, key = lamda x: x['instance_id']

```

File: luxurynitro/api.py [hunk 4]
```diff
@@ -79,6 +79,20 @@ async def get_user(self) -> classes.User:
...
+        res = await self.client.get('/vps-stats')
+        resjson = res.json()
+
+        return [
+            classes.VPS(
+                alts = instance['alts'],
+                instance_id = instance["instance_id"],
+                last_seen = instance["last_seen"],
+                servers = instance["servers"]
+            ) for instance in resjson
```
Comment (itschasa) [inline luxurynitro/api.py:93]:
```suggestion
            ) for instance in data

```

**Review by itschasa — CHANGES_REQUESTED** [review]
have a look at my suggested changes

**Review by InvalidLenni — COMMENTED** [review]
idk if its needed lol

File: main.py [hunk 5]
```diff
@@ -343,7 +348,60 @@ async def on_ready():
...
+    except luxurynitro.errors.APIError as exc:
+        await log.warn(f"{utils.lang.embed_fetch_error} {exc.message}")
+    except luxurynitro.errors.RetryTimeout as exc:
+        await log.warn(f"{utils.lang.embed_fetch_error} {exc.message}" + "\n- ".join(f"`{e}`" for e in exc.errors))
+    else:
+        vps_stats_sorted = sorted(vps_stats, key=lambda x: x.instance_id)
+
+        extensions = '\n'.join(
+            [
+                f'{config.vps_webhook.emojis["working"]} ``Instance {stats.instance_id}`` - ``{stats.servers} Guilds / {stats.alts} alts``'
```
Comment (itschasa) [inline main.py:368]:
```suggestion
                f'{config.vps_webhook.emojis["offline"] if stats.last_seen < current_time - 45 else config.vps_webhook.emojis["online"]} ``Instance {stats.instance_id}`` - ``{stats.servers} Guilds / {stats.alts} alts``'

```

File: main.py [hunk 6]
```diff
@@ -343,7 +348,60 @@ async def on_ready():
...
+    global vps_message_id, last_update_ping
+    await client.wait_until_ready()
+    try:
+        vps_stats = await api.get_vps_stats()
+    except luxurynitro.errors.APIError as exc:
+        await log.warn(f"{utils.lang.embed_fetch_error} {exc.message}")
+    except luxurynitro.errors.RetryTimeout as exc:
+        await log.warn(f"{utils.lang.embed_fetch_error} {exc.message}" + "\n- ".join(f"`{e}`" for e in exc.errors))
+    else:
+        vps_stats_sorted = sorted(vps_stats, key=lambda x: x.instance_id)
```
Comment (itschasa) [inline main.py:364]:
```suggestion
        current_time = int(time.time())
        vps_stats_sorted = sorted(vps_stats, key=lambda x: x.instance_id)
```

**Review by itschasa — COMMENTED** [review]
i will commit these changes

**Review by itschasa — APPROVED** [review]

**Pull request merged**